src/darwin/Framework/CHIP/CHIPDevice.mm

```diff
@@ -173,6 +202,898 @@ - (void)subscribeWithQueue:(dispatch_queue_t)queue
+            VerifyOrReturnError(commandSender != nullptr, CHIP_ERROR_NO_MEMORY);
+
+            ReturnErrorOnFailure(commandSender->AddRequestData(commandPath, NSObjectData(commandFields), chip::NullOptional));
+            ReturnErrorOnFailure(commandSender->SendCommandRequest([self internalDevice]->GetSecureSession().Value()));
+
+            decoder.release();
+            commandSender.release();
+            return CHIP_NO_ERROR;
+        });
+}
+
+- (void)subscribeAttributeWithEndpointId:(NSUInteger)endpointId
+                               clusterId:(NSUInteger)clusterId
+                             attributeId:(NSUInteger)attributeId
+                             minInterval:(NSUInteger)minInterval
+                             maxInterval:(NSUInteger)maxInterval
+                             clientQueue:(dispatch_queue_t)clientQueue
+                           reportHandler:(void (^)(NSDictionary<NSString *, id> * _Nullable value,
+                                             NSError * _Nullable error))reportHandler
+                 subscriptionEstablished:(SubscriptionEstablishedHandler)subscriptionEstablishedHandler
+{
+    new NSObjectAttributeCallbackSubscribeBridge(
+        clientQueue, (void (^)(id _Nullable, NSError * _Nullable)) reportHandler,
+        ^(chip::Callback::Cancelable * success, chip::Callback::Cancelable * failure) {
+            auto successFn = chip::Callback::Callback<NSObjectAttributeCallback>::FromCancelable(success);
+            auto failureFn = chip::Callback::Callback<CHIPErrorCallback>::FromCancelable(failure);
+            auto context = successFn->mContext;
+            auto successCb = successFn->mCall;
+            auto failureCb = failureFn->mCall;
+            auto onReportCb = [context, successCb](const app::ConcreteAttributePath & attribPath, const NSObjectData & data) {
+                if (successCb != nullptr) {
+                    successCb(context,
+                        [[NSDictionary alloc] initWithObjectsAndKeys:[NSNumber numberWithUnsignedInt:attribPath.mEndpointId],
+                                              kCHIPEndpointIdKey, [NSNumber numberWithUnsignedInt:attribPath.mClusterId],
+                                              kCHIPClusterIdKey, [NSNumber numberWithUnsignedInt:attribPath.mAttributeId],
+                                              kCHIPAttributeIdKey, data.GetDecodedObject(), kCHIPValueKey, nil]);
+                }
+            };
+
+            auto establishedOrFailed = std::make_shared<BOOL>(NO);
+            auto onFailureCb
+                = [context, failureCb, establishedOrFailed](const app::ConcreteAttributePath * attribPath, CHIP_ERROR error) {
+                      if (!(*establishedOrFailed)) {
+                          *establishedOrFailed = YES;
+                          NSObjectAttributeCallbackSubscribeBridge::OnSubscriptionEstablished(context);
```

Comment:
Wait, that's not how it works for the other subscription APIs, and this API does not document this behavior...

That said, I think we need to change some of the other subscription APIs around this stuff, so we should probably revisit this API after that.